cpp/tests/io/orc_test.cpp

```diff
@@ -1264,6 +1264,60 @@ TEST_F(OrcStatisticsTest, Overflow)
   check_sum_exist(3, true);
   check_sum_exist(4, true);
 }
+
+TEST_F(OrcStatisticsTest, HasNull)
+{
+  // cudf's ORC writer doesn't yet support the ability to encode the hasNull value in statistics so
+  // we're embedding a file created using pyorc
+  //
+  // Method to create file:
+  // >>> import pyorc
+  // >>> output = open("./temp.orc", "wb")
+  // >>> writer = pyorc.Writer(output, pyorc.Struct(a=pyorc.BigInt(), b=pyorc.BigInt()))
+  // >>> writer.write((1, 3))
+  // >>> writer.write((2, 4))
+  // >>> writer.write((None, 5))
+  // >>> writer.close()
+  //
+  // Contents of file:
+  // >>> import pyarrow.orc as po
+  // >>> po.ORCFile('new.orc').read()
+  // pyarrow.Table
+  // a: int64
+  // b: int64
+  // ----
+  // a: [[1,2,null]]
+  // b: [[3,4,5]]
+  auto nulls_orc = std::array<uint8_t, 308>{
```

Comment:
If this is going to be replaced by a round-trip write+read in the future then we may need to add comment to clarify that here.